### DISCUSSION TITLE
Debug Python CDK app

### DISCUSSION THREAD

**Discussion by avoidik** [microsoft/debugpy#1123] [Q&A]
Hello,

I'm trying to debug my [Python CDK app](https://docs.aws.amazon.com/cdk/v2/guide/getting_started.html) via VSCode using [this article](https://code.visualstudio.com/docs/python/debugging) as my how-to guide. Everything works as intended however the process being debugged is unable to access `cdk.json` file from the current project's directory, this is the special file used to define [a runtime context](https://docs.aws.amazon.com/cdk/v2/guide/context.html) together with a bunch of run-time variables. If I run my process outside the debugging session then this file becomes magically accessible. It sounds like the working directory was set to something where my code cannot find the `cdk.json` file. I was wondering I'm missing something, any tips here?

**Comment by int19h** [comment]
If your launch.json has `"program": "..."` with absolute path, that path is used as the current directory by default. However, you can override that by specifying `"cwd": "..."` as needed.

**Reply by avoidik** [reply]
thanks Pavel for your suggestion, unfortunately it doesn't work, here is my `launch.json` config file I am using in vscode to debug my application 

```json
{
    // Use IntelliSense to learn about possible attributes.
    // Hover to view descriptions of existing attributes.
    // For more information, visit: https://go.microsoft.com/fwlink/?linkid=830387
    "version": "0.2.0",
    "configurations": [
        {
            "name": "Python: Debug cdk",
            "type": "python",
            "request": "launch",
            "stopOnEntry": false,
            "module": "demo_app.app",
            "console": "integratedTerminal",
            "justMyCode": false,
            "cwd": "${workspaceFolder}"
        },
    ]
}
```

however if I set `env` as follows it works

```json
"env": {
  "CDK_CONTEXT_JSON": "{\"aaa\": \"bbb\", \"ccc\": \"ddd\"}"
}
```

I am still trying to understand whether this specific behavior corresponds to debugpy or cdk itself, I am going to ask CDK community the same question
